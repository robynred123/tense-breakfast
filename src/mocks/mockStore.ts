import configureStore from 'redux-mock-store';
import { Store, AnyAction } from 'redux';
import { FilterOptions, ReducerState } from '../types/index';

export const filterOptions: FilterOptions = {
  appointmentType: [],
  dateRange: {
    start: null,
    end: null,
  },
  specialisms: [],
};
export const initialState: ReducerState = {
  therapists: [],
  filteredTherapists: [],
  filterOptions,
  mobileFilter: false,
  error: null,
};

export const mockStore = (initialState: ReducerState) => {
  const mockStore = configureStore();
  let store: Store<any, AnyAction>;
  store = mockStore({ one: initialState });
  return store;
};
