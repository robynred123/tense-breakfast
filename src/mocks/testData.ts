import { TherapistInfo } from '../types';

export const therapistOne: TherapistInfo = {
  id: '063eb398-45f7-4a9c-a5d3-e95d14c05c44',
  firstName: 'Albert',
  lastName: 'Einstein',
  appointment_types: ['one_off'],
  appointment_mediums: ['phone'],
  specialisms: ['Alcoholism', 'Gender identity', 'Gambling Addiction'],
};

export const therapistTwo: TherapistInfo = {
  id: '154aa58b-4c25-4d64-a7a2-9288a7d94468',
  firstName: 'Jackie',
  lastName: 'Chan',
  appointment_types: ['consultation', 'one_off'],
  appointment_mediums: ['video'],
  specialisms: ['Gender identity', 'Alcoholism'],
};
