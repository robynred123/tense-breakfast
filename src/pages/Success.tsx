import { Box, Grid, Typography } from '@mui/material';
import { useNavigate } from 'react-router-dom';
import { ButtonComponent } from '../components/Button';

export const SuccessPage = () => {
  const navigate = useNavigate();
  return (
    <Box sx={{ flexGrow: 1, width: 1, flexDirection: 'row', wrap: 'nowrap' }}>
      <Grid
        item
        md={12}
        lg={12}
        sm={12}
        sx={{
          padding: '10%',
          justifyContent: 'center',
          textAlign: 'center',
        }}
      >
        <Typography variant='h2' fontFamily={'lato, sans-serif'} sx={{ paddingBottom: '1em' }}>
          Appointment Requested
        </Typography>
        <Typography variant='h3' fontFamily={'lato, sans-serif'}>
          You will receive confirmation shortly
        </Typography>
      </Grid>
      <Grid
        item
        lg={12}
        sx={{
          justifyContent: 'center',
          textAlign: 'center',
        }}
      >
        <ButtonComponent
          onClick={() => navigate('/')}
          text='OK'
          buttonColour={'gradient'}
          disabled={false}
          width={'20%'}
        />
      </Grid>
    </Box>
  );
};
