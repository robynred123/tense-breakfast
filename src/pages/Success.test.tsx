/* eslint-disable testing-library/await-async-query */
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { mockStore, initialState } from '../mocks/mockStore';
import { SuccessPage } from './Success';
import { Router } from 'react-router';
import { createBrowserHistory } from 'history';

describe('<SuccessPage />', () => {
  let mockUseLocationValue = {
    pathname: '/Success',
    search: '',
    hash: '',
    state: null,
  };

  const history = createBrowserHistory();

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('Should render the Success Page successfully', () => {
    const store = mockStore(initialState);
    const view = renderer.create(
      <Provider store={store}>
        <Router navigator={history} location={'/'}>
          <SuccessPage />
        </Router>
      </Provider>
    );
    expect(view.toJSON()).toMatchSnapshot();
    expect(view.root.findByType('h2').props.children).toEqual('Appointment Requested');
    expect(view.root.findByType('h3').props.children).toEqual(
      'You will receive confirmation shortly'
    );
    expect(view.root.findByType('button').props.children).toEqual([
      [undefined, 'OK', undefined],
      null,
    ]);
  });
});
