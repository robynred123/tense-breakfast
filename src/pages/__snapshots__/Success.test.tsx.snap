// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuccessPage /> Should render the Success Page successfully 1`] = `
<div
  className="MuiBox-root css-1rq3dmv"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 css-dmqau3-MuiGrid-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2 css-a4vnku-MuiTypography-root"
    >
      Appointment Requested
    </h2>
    <h3
      className="MuiTypography-root MuiTypography-h3 css-vj3jzm-MuiTypography-root"
    >
      You will receive confirmation shortly
    </h3>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12 css-1aiqtfq-MuiGrid-root"
  >
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-cq6rpw-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      OK
    </button>
  </div>
</div>
`;
